flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/DeclarativeScheduler.java

```diff
@@ -207,6 +208,12 @@ public DeclarativeScheduler(
                         declarativeSlotPool::reserveFreeSlot,
                         declarativeSlotPool::freeReservedSlot);
 
+        for (JobVertex vertex : jobGraph.getVertices()) {
+            if (vertex.getParallelism() == ExecutionConfig.PARALLELISM_DEFAULT) {
+                vertex.setParallelism(1);
+            }
+        }
```

Comment:
How is the `DefaultScheduler` handling this situation?

Reply:
The ExecutionJobVertex checks the set parallelism, and if it is negative sets it to a default parallelism that is also passed via the constructor. This argument is always `1`.

Reply:
The ExecutionJobVertex constructor sets it if it is negative to a default parallelism that is also passed via the constructor, which is always `1`.

Reply:
Alright, thanks for the clarification.